sandbox/modules/identity/src/main/java/org/opensearch/identity/realm/InternalRealm.java

```diff
@@ -153,137 +139,10 @@ protected AuthenticationInfo doGetAuthenticationInfo(AuthenticationToken token)
         throw new CredentialsException();
     }
 
-    // TODO: Expose all the operations below as a rest API
```

Comment:
I see you dropped a lot of these steps in this file and the InternalRealmTests file. Are you planning to re-add some of this functionality later? For instance, I see that there is no longer a removeAttributesFromUsers method--is this something that needs to be reimplemented or am I just missing it.

Reply:
Take a look at this example from the security plugin (`saveAnUpdateConfigs` - yes there is a typo in that name): https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/dlic/rest/api/AbstractApiAction.java#L300

After a successful update to the `.opendistro_security` index (which is the index that persists the security config inside opensearch) it calls on a `ConfigUpdateAction`: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/dlic/rest/api/AbstractApiAction.java#L333-L353

That config update action calls on `reloadConfiguration` in here within the `nodeOperation()`: https://github.com/opensearch-project/security/blob/main/src/main/java/org/opensearch/security/action/configupdate/TransportConfigUpdateAction.java#L107-L112.

The `TransportConfigUpdateAction` subclasses `TransportNodesAction` which indicates that this action runs on all nodes.

This `reloadConfiguration` is central to how security works in a distributed way and make sure that all nodes in the cluster stay in sync.

Inside `reloadConfiguration` it queries the system index and loads the configuration into memory and uses an `eventBus` to publish to subscribing classes changes in models that they rely on. In this PR, `InternalUsersStore` is passed to `registerDCFListener` (DCF == DynamicConfigFactory) so its a class that subscribes to changes on the `InternalUsersModel`.

**tldr;** Changes to a user should directly be saved into the system index and `reloadConfiguration` will take care of the rest to ensure the nodes of the cluster stay in sync.

Reply:
@DarshitChanpura's PR to create a user via API uses this system to propagate security changes to all nodes of the cluster: https://github.com/opensearch-project/OpenSearch/pull/5990

Reply:
Thank you for the detailed explanation. It is great that you caught this then since looks like there was a lot of extra leg work being done. Everything looks good to me!